test/FeeOnTransfer.spec.ts

```diff
@@ -0,0 +1,119 @@
+import { expect } from 'chai'
+import { constants } from 'ethers'
+import hre, { ethers } from 'hardhat'
+import { FeeOnTransfer, TestERC20 } from '../typechain'
+
+describe('FeeOnTransfer', function () {
+  let feeOnTransfer: FeeOnTransfer
+  let testToken: TestERC20
+
+  // WETH9 and USDC
+  const BASE_TOKENS = ['0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2', '0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48']
+  // Arbitrary amount to flash loan.
+  const AMOUNT_TO_BORROW = 1000
+
+  const FOT_TOKENS = [
+    '0xa68dd8cb83097765263adad881af6eed479c4a33', // WTF
+    '0x8B3192f5eEBD8579568A2Ed41E6FEB402f93f73F', // SAITAMA
+    '0xA2b4C0Af19cC16a6CfAcCe81F192B024d625817D', // KISHU
+  ]
+
+  const NON_FOT_TOKENS = [
+    '0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48', // USDC
+    '0x1f9840a85d5aF5bf1D1762F925BDADdC4201F984', // UNI
+    '0xc00e94Cb662C3520282E6f5717214004A7f26888', // COMP
+    '0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2', // WETH9
+  ]
+
+  before(async function () {
```

Comment:
if you don't want to go through all this mainnet forking stuff it wouldn't be _too_ hard to mock up something like https://github.com/Uniswap/v2-periphery/blob/master/contracts/test/DeflatingERC20.sol and test this statically

fine with it as is though

Reply:
Will leave it as is, I quite like that we test against the tokens that users have reported having issues with so we're 100% confident it will fix the issue